ch5/exercise5-9/expand_test.go

```diff
@@ -0,0 +1,31 @@
+package main
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestExpand(t *testing.T) {
+
+	var tests = []struct {
+		input  string
+		num    uint
+		output string
+	}{
+		{"hi", 4, "hi"},
```

Comment:
You may even consider precomputing timesX.  E.g.

```go
var tests = []struct {
    input  string
    fn     expandFn
    output string
} {
    {"$hi", timesX(3), "hihihi"},
}
```

No need to change it now.  Just further illustration of functions as data.